content-src/lib/selectLayoutRender.js

```diff
@@ -30,8 +30,10 @@ export const selectLayoutRender = (state, prefs, rickRollCache) => {
 
       if (rickRoll <= spocsConfig.probability) {
         spocIndex++;
-        recommendations.splice(position.index, 0, spoc);
-        chosenSpocs.add(spoc);
+        if (!spocs.blocked.includes(spoc.url)) {
+          recommendations.splice(position.index, 0, spoc);
+          chosenSpocs.add(spoc);
+        }
```

Comment:
Including @ncloudioj to verify change here of increment spocIndex with possibility of not updating chosenSpocs impact outOfPositionSpocsFill telemetry on line 203 below

Reply:
Confirmed that `spocIndex++` is still necessary here regardless whether or not this spoc was blocked.  See [this comment](https://github.com/mozilla/activity-stream/pull/5188/files#diff-91e34b3436d635509e2017c193c45ec1R185) for more details.

Reply:
Thanks @ncloudioj for clarifying, so its safe to assume spocsFill telemetry is not impacted with above change and we should increment and keep spocIndex++